processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessPipeline.java

```diff
@@ -60,28 +60,49 @@ public ProcessPipeline(ThreadScope scope,
         processMetrics = metrics.new ProcessMetrics();
     }
 
-    public CompletableFuture<Void> scheduleThenProcess(TaskRequest request) throws InterruptedException {
-        DecatonTask<T> extracted = extract(request);
+    public void scheduleThenProcess(TaskRequest request) throws InterruptedException {
+        final DecatonTask<T> extracted;
+        try {
+            extracted = extract(request);
+        } catch (RuntimeException e) {
+            log.error("Dropping not-deserializable task [{}]", request.id(), e);
+            taskMetrics.tasksDiscarded.increment();
+            return;
+        }
 
         scheduler.schedule(extracted.metadata());
+        if (terminated) {
+            log.debug("Returning after schedule due to termination");
+            return;
+        }
 
-        return process(request, extracted);
+        CompletableFuture<Void> result = process(request, extracted);
+        DeferredCompletion completion = request.completion();
+        result.whenComplete((r, e) -> {
+            if (e != null) {
+                if (e instanceof InterruptedException && true /* terminated */) {
```

Comment:
`&& true` ?

Reply:
as I mentioned briefly, I was still considering whether this special treatment worth it to keep or not. I'll make a decision and fix it soon.